src/happy.rs

```diff
@@ -0,0 +1,84 @@
+//! aaa
+use std::io;
+use std::iter::FusedIterator;
+use std::net::TcpStream;
+use std::net::ToSocketAddrs;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::mpsc::channel;
+use std::sync::Arc;
+use std::thread;
+use std::time::{Duration, Instant};
+
+fn intertwine<T, A, B>(mut ita: A, mut itb: B) -> Vec<T>
```

Comment:
It is definitely a nit, but at work we usually begin a module with the public API and have the implementation details at the bottom so it is easier for an unfamiliar reader to get the gist of a module without being swamped by helper types and functions.

Reply:
This is a WIP so it's a bit messy. I might move this to a util module or something.